### PR TITLE
fix(Core/Unit): Implement GetDisplayRace()

### PR DESCRIPTION
**Pull request by sudlud**
## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes 

- Prerequisite for #17632 

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by kissingers — APPROVED** [review]

**Review by Kitzunu — APPROVED** [review]

File: src/server/shared/SharedDefines.h [hunk 1]
```diff
@@ -105,6 +105,33 @@ enum Races
 
 #define RACEMASK_HORDE RACEMASK_ALL_PLAYABLE & ~RACEMASK_ALLIANCE
 
+// DisplayRace values from CreatureDisplayInfoExtra.dbc
+enum class DisplayRace : uint8
```
Comment (natrist) [inline src/server/shared/SharedDefines.h:109]: Constants should be in uppercase.

**Review by natrist — CHANGES_REQUESTED** [review]

File: src/server/shared/SharedDefines.h (hunk 1 above)
Comment (sudlud) [reply src/server/shared/SharedDefines.h:109]: I used #17607 by @Kitzunu as a reference which also has no upper case constants in enum class

Comment (natrist) [reply src/server/shared/SharedDefines.h:109]: > I used #17607 by @Kitzunu as a reference which also has no upper case constants in enum class

Nothing is sacred anymore 😭

Comment (Kitzunu) [reply src/server/shared/SharedDefines.h:109]: In enum we use uppercase, but not in enum class

**Comment by sudlud** [comment]
@Kitzunu is there anything to to from my side here? 🙃 
Follow-up PR is already ready and waiting 👀😃

**Comment by sudlud** [comment]
Anything to to here from my side? 🙃

**Pull request merged**